src/components/InjectedComponents/PostDialog.tsx

```diff
@@ -89,21 +98,41 @@ export interface PostDialogUIProps
     onlyMyself: boolean
     shareToEveryone: boolean
     imagePayload: boolean
+    imageEncrypt: boolean
     availableShareTarget: Array<Profile | Group>
     currentShareTarget: Array<Profile | Group>
     currentIdentity: Profile | null
     postContent: TypedMessage
+    onImgChange: (img: File) => Promise<void>
     postBoxButtonDisabled: boolean
     onPostContentChanged: (nextMessage: TypedMessage) => void
     onOnlyMyselfChanged: (checked: boolean) => void
     onShareToEveryoneChanged: (checked: boolean) => void
     onImagePayloadSwitchChanged: (checked: boolean) => void
+    onImageEncryptSwitchChanged: (checked: boolean) => void
     onFinishButtonClicked: () => void
     onCloseButtonClicked: () => void
     onSetSelected: SelectRecipientsUIProps['onSetSelected']
     DialogProps?: Partial<DialogProps>
     SelectRecipientsUIProps?: Partial<SelectRecipientsUIProps>
 }
+
+// TODO: you will want to move this elsewhere
+const readFileAsync: (file: File) => Promise<ArrayBuffer | null> = (file: File) => {
```

Comment:
Looks familiar, we have the same logic at somewhere else. we can make it into utils